src/Lean/Meta/ExprDefEq.lean

```diff
@@ -728,7 +728,7 @@ mutual
     else
       let lctx := ctxMeta.lctx
       match lctx.findFVar? fvar with
-      | some (.ldecl (value := v) ..) => check v
+      | some (.ldecl (nondep := false) (value := v) ..) => check v
       | _ =>
```

Comment:
@kmill Could you please document this change? We need a `why?` and example showing why the previous behavior was incorrect.

Reply:
Added an explanation and example.